src/resolution-map.js

```diff
@@ -48,7 +49,7 @@ export default class ResolutionMap {
   }
 
   parsePatternInfo(globPattern: string, range: string): ?Object {
-    const directories = globPattern.split(DIRECTORY_SEPARATOR);
+    const directories = parsePackagePath(globPattern);
     const name = directories.pop();
 
     if (!name) {
```

Comment:
I'd still prefer the regex version -- you can catch the name error by doing something like
```
if (globPattern.match(/(\/|\*+)$/)) {
  this.reporter.warn(this.reporter.lang('invalidResolutionName', globPattern));
}

const directories = parsePackagePath(globPattern); // input.match(/(@[^\/]+\/)?([^/]+)/g);
const name = directories.pop();
```

this way we can catch invalid names beyond just trailing slash, and in general, less line of code is easier to maintain :)